src/main/java/org/ays/auth/service/impl/AysUserUpdateServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+package org.ays.auth.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.ays.auth.model.AysRole;
+import org.ays.auth.model.AysUser;
+import org.ays.auth.model.request.AysUserUpdateRequest;
+import org.ays.auth.port.AysRoleReadPort;
+import org.ays.auth.port.AysUserReadPort;
+import org.ays.auth.port.AysUserSavePort;
+import org.ays.auth.service.AysUserUpdateService;
+import org.ays.auth.util.exception.AysRolesNotExistException;
+import org.ays.auth.util.exception.AysUserNotExistByIdException;
+import org.ays.auth.util.exception.AysInvalidUserStatusException;
+import org.ays.auth.util.exception.AysPhoneNumberAlreadyInUseException;
+import org.ays.auth.util.exception.AysEmailAlreadyInUseException;
+import org.ays.common.model.AysPhoneNumber;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Service implementation for updating users.
+ * This service handles the update operation of existing users based on the provided update request.
+ */
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class AysUserUpdateServiceImpl implements AysUserUpdateService {
```

Comment:
Implementasyonlara diğer sınıflardan erişmemek için package-private yapmamızda fayda var
```suggestion
class AysUserUpdateServiceImpl implements AysUserUpdateService {
```

Reply:
değiştirdim